bolts/data/datamodules/base.py

```diff
@@ -50,8 +50,8 @@ def __init__(
         self.stratified_sampling = stratified_sampling
         self.instance_weighting = instance_weighting
         self.training_mode = training_mode
-        self._s_dim = None
-        self._y_dim = None
+        self._card_s = None
+        self._card_y = None
         self._train_data = None
```

Comment:
Can the other splits also be moved inside the init and have their value initially set to `None`?